cpp/src/arrow/acero/swiss_join_avx2.cc

```diff
@@ -45,48 +45,62 @@ int RowArrayAccessor::Visit_avx2(const RowTableImpl& rows, int column_id, int nu
   if (!is_fixed_length_column) {
     int varbinary_column_id = VarbinaryColumnId(rows.metadata(), column_id);
     const uint8_t* row_ptr_base = rows.data(2);
-    const uint32_t* row_offsets = rows.offsets();
+    const RowTableImpl::offset_type* row_offsets = rows.offsets();
```

Comment:
Just found that the functions in this file are not used anywhere. I updated it still though - maybe I'll clean it up or make it useful in the future.

Reply:
Hmm, do you mean the AVX2 speedups are actually not wired anywhere?

Reply:
Right, they are not as far as I searched the code.